AutoGraph/Request.swift

```diff
@@ -1,54 +1,22 @@
-import Crust
 import Foundation
 import JSONValueRX
 
 /// A `Request` to be sent by AutoGraph.
 public protocol Request {
-    /// The `Mapping` used to map from the returned JSON payload to a concrete type
-    /// `Mapping.MappedObject`.
-    associatedtype Mapping: Crust.Mapping
-    
-    /// The root key into the JSON payload. E.g. If the payload has `[ "data" : [ stuff_I_want ] ]` the `RootKey`
-    /// will be a `String` and the instance will be `"data"`.
-    associatedtype RootKey: MappingKey
-    
-    /// The keys we'll use to map data out of the JSON payload. This payload starts from the JSON retreived from the root key in `mapping`.
-    associatedtype MappingKeys: KeyCollection where MappingKeys.MappingKeyType == Mapping.MappingKeyType
-    
     /// The returned type for the request.
-    /// E.g if the requests returns an array then change to `[Mapping.MappedObject]`.
-    associatedtype SerializedObject = Mapping.MappedObject
+    associatedtype SerializedObject: Decodable
     
     associatedtype QueryDocument: GraphQLDocument
     associatedtype Variables: GraphQLVariables
     
-    /// If the `SerializedObject`(s) cannot be passed across threads, then we'll use this to transform
-    /// the objects as they are passed from the background to the main thread.
-    associatedtype ThreadAdapterType: ThreadAdapter
-    
     /// The query to be sent to GraphQL.
     var queryDocument: QueryDocument { get }
     
     /// The variables sent along with the query.
     var variables: Variables? { get }
     
-    /// The mapping to use when mapping JSON into the a concrete type.
-    ///
-    /// **WARNING:**
-    ///
-    /// `mapping` does NOT execute on the main thread. It's important that any `Adapter`
-    /// used by `mapping` establishes its own connection to the DB from within `mapping`.
-    ///
-    /// Additionally, the mapped data (`Mapping.MappedObject`) is assumed to be safe to pass
-    /// across threads unless it inherits from `ThreadUnsafe`.
-    var mapping: Binding<RootKey, Mapping> { get }
-    
-    /// The collection of keys / fields that will be mapped from the query for this request.
-    var mappingKeys: MappingKeys { get }
-    
-    /// Our `ThreadAdapter`. Its `typealias BaseType` must be a the same type or a super type of `Mapping.MappedObject`
-    /// or an error will be thrown at runtime.
-    var threadAdapter: ThreadAdapterType? { get }
+    /// The key path to the result object in the data
+    var objectRootKeyPath: String { get }
```

Comment:
Replaces the `RootKey` requirement.

Reply:
`object` sounds too ambiguous, either `serializedObjectRootKeyPath` or `serializedObjectRootKey` or just go for brevity and use `rootKey`.